src/Runner.jl

```diff
@@ -126,10 +126,11 @@ function generate_compiler_wrappers!(platform::AbstractPlatform; bin_path::Abstr
             println(io)
         end
 
-        # If we're given compile-only flags, include them only if `-x assembler` is not provided
+        # If we're given compile-only flags, include them only if `-x assembler` is not provided,
+        # or no object files (*.o) are passed in input
         if !isempty(compile_only_flags)
             println(io)
-            println(io, "if [[ \" \${ARGS[@]} \" != *' -x assembler '* ]]; then")
+            println(io, "if [[ \" \${ARGS[@]} \" != *' -x assembler '* ]] && [[ \" \${ARGS[@]} \" != *'.o '* ]]; then")
```

Comment:
This seems like a hard problem to solve.  I'm honestly not sure what to do here.

On the one hand, we need a way to tell `clang` that it should be in `-x objc` mode, but it throws an error if we're linking.....  On modern macs, the `objc` executable doesn't even exist, so perhaps this whole compiler wrapper is just a bad idea.  What software are you building that needs this?

Reply:
Meson needs it...

Reply:
So, this is the problem: we need to specify an Objective-C compiler in the Meson cross-file:
https://github.com/JuliaPackaging/BinaryBuilderBase.jl/blob/cd3f4d129c4fe5126b13a57979a30c927ae2f0a0/src/BuildToolchains.jl#L167
Meson is very strict in many ways, I'm 99% sure it won't accept something like `clang -x objective-c`, and anyways it uses the compiler also as linker (which is what's happening in https://github.com/JuliaPackaging/Yggdrasil/pull/91#issuecomment-842704490).

Reply:
Uhm, let me see if I can do it with
```python
objc = '/opt/bin/$(target)/$(aatarget)-clang'
# ...
objc_args = ['-x objective-c']
```